modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/url/FDSAPIURLBuilderImpl.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set.internal.url;
+
+import com.liferay.frontend.data.set.url.FDSAPIURLBuilder;
+import com.liferay.frontend.data.set.url.FDSAPIURLResolver;
+import com.liferay.frontend.data.set.url.FDSAPIURLResolverRegistry;
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Daniel Sanz
+ */
+public class FDSAPIURLBuilderImpl implements FDSAPIURLBuilder {
+
+	public FDSAPIURLBuilderImpl(
+		String restApplication, String restEndpoint, String restSchema,
+		FDSAPIURLResolverRegistry fdsAPIURLResolverRegistry,
+		HttpServletRequest httpServletRequest) {
+
+		_restApplication = restApplication;
+		_restEndpoint = restEndpoint;
+		_restSchema = restSchema;
+		_fdsAPIURLResolverRegistry = fdsAPIURLResolverRegistry;
+		_httpServletRequest = httpServletRequest;
+	}
+
+	@Override
+	public FDSAPIURLBuilder addParameter(String name, String value) {
+		if (Validator.isNotNull(name) && Validator.isNotNull(value)) {
+			_queryStringItems.add(name + CharPool.EQUAL + value);
```

Comment:
I _think_ three concatenating items with + is acceptable, four would require a builder.

Reply:
SF did not complain, yep